pkg/core/native_contract_test.go

```diff
@@ -1,324 +1,168 @@
-		v.Estack().PushVal(14)
-		v.Estack().PushVal(28)
-		v.Context().Jump(sumOffset)
+		v.LoadScriptWithHash(clState.NEF.Script, clState.Hash, callflag.All)
+		input := []byte{1, 2, 3, 4}
+		v.Estack().PushVal(input)
+		v.Context().Jump(md.Offset)
+
 		require.NoError(t, v.Run())
 
-		value := v.Estack().Pop().BigInt()
-		require.Equal(t, int64(42), value.Int64())
+		value := v.Estack().Pop().Bytes()
+		require.Equal(t, hash.RipeMD160(input).BytesBE(), value)
 	})
 }
 
 func TestNativeContract_InvokeOtherContract(t *testing.T) {
-	chain := newTestChain(t)
-
-	tn := newTestNative()
-	chain.registerNative(tn)
-
-	err := chain.contracts.Management.PutContractState(chain.dao, &state.Contract{
-		ContractBase: state.ContractBase{
-			ID:       1,
-			Hash:     tn.meta.Hash,
-			NEF:      tn.meta.NEF,
-			Manifest: tn.meta.Manifest,
-		},
-	})
+	bc, validator, committee := chain.NewMulti(t)
+	e := neotest.NewExecutor(t, bc, validator, committee)
+	managementInvoker := e.ValidatorInvoker(e.NativeHash(t, nativenames.Management))
+	gasInvoker := e.ValidatorInvoker(e.NativeHash(t, nativenames.Gas))
+
+	cs, _ := contracts.GetTestContractState(t, pathToInternalContracts, 1, 2, validator.ScriptHash())
+	cs.Hash = state.CreateContractHash(validator.ScriptHash(), cs.NEF.Checksum, cs.Manifest.Name) // set proper hash
+	manifB, err := json.Marshal(cs.Manifest)
 	require.NoError(t, err)
+	nefB, err := cs.NEF.Bytes()
+	require.NoError(t, err)
+	si, err := cs.ToStackItem()
+	require.NoError(t, err)
+	managementInvoker.Invoke(t, si, "deploy", nefB, manifB)
 
-	var drainTN = func(t *testing.T) {
-		select {
-		case <-tn.blocks:
-		default:
-			require.Fail(t, "testNative didn't send us block")
-		}
-	}
-
-	cs, _ := contracts.GetTestContractState(t, pathToInternalContracts, 4, 5, random.Uint160()) // sender and IDs are not important for the test
-	require.NoError(t, chain.contracts.Management.PutContractState(chain.dao, cs))
-
-	baseFee := chain.GetBaseExecFee()
 	t.Run("non-native, no return", func(t *testing.T) {
-		res, err := invokeContractMethod(chain, testSumCPUFee*baseFee*4+10000, tn.Metadata().Hash, "callOtherContractNoReturn", cs.Hash, "justReturn", []interface{}{})
-		require.NoError(t, err)
-		drainTN(t)
-		require.Equal(t, vm.HaltState, res.VMState, res.FaultException)
-		checkResult(t, res, stackitem.Null{}) // simple call is done with EnsureNotEmpty
+		// `onNEP17Payment` will be invoked on test contract from GAS contract.
+		gasInvoker.Invoke(t, true, "transfer", validator.ScriptHash(), cs.Hash, 1, nil)
 	})
 	t.Run("non-native, with return", func(t *testing.T) {
-		res, err := invokeContractMethod(chain, testSumCPUFee*baseFee*4+10000, tn.Metadata().Hash,
-			"callOtherContractWithReturn", cs.Hash, "ret7", []interface{}{})
-		require.NoError(t, err)
-		drainTN(t)
-		checkResult(t, res, stackitem.Make(8))
+		// TODO: we don't have a case where native contract calls non-native contract with return value.
```

Comment:
@roman-khimov, can we remove this test?

Reply:
The problem is that there's no case when native contract calls non-native contract with return. How can I test it if all our native contracts can call only those non-native contracts that do not have return value?

Reply:
Ah, OK then.

Reply:
Removed.